libs/libc/machine/arm64/arch_elf.c

```diff
@@ -0,0 +1,794 @@
+
+  /* Verify endian-ness */
+
+#ifdef CONFIG_ENDIAN_BIG
+  if (ehdr->e_ident[EI_DATA] != ELFDATA2MSB)
+#else
+  if (ehdr->e_ident[EI_DATA] != ELFDATA2LSB)
+#endif
+    {
+      berr("ERROR: Wrong endian-ness: e_ident[EI_DATA]=%02x\n",
+           ehdr->e_ident[EI_DATA]);
+      return false;
+    }
+
+  /* TODO:  Check ABI here. */
+
+  return true;
+}
+
+/****************************************************************************
+ * Name: up_relocate and up_relocateadd
+ *
+ * Description:
+ *   Perform an architecture-specific ELF relocation.  Every architecture
+ *   that uses the ELF loader must provide this function.
+ *
+ * Input Parameters:
+ *   rel - The relocation type
+ *   sym - The ELF symbol structure containing the fully resolved value.
+ *         There are a few relocation types for a few architectures that do
+ *         not require symbol information.  For those, this value will be
+ *         NULL.  Implementations of these functions must be able to handle
+ *         that case.
+ *   addr - The address that requires the relocation.
+ *
+ * Returned Value:
+ *   Zero (OK) if the relocation was successful.  Otherwise, a negated errno
+ *   value indicating the cause of the relocation failure.
+ *
+ ****************************************************************************/
+
+int up_relocate(const Elf64_Rel *rel, const Elf64_Sym *sym, uintptr_t addr)
+{
+  berr("ERROR: REL relocation not supported\n");
+  return -ENOSYS;
+}
+
+int up_relocateadd(const Elf64_Rela *rel, const Elf64_Sym *sym,
+                   uintptr_t addr)
+{
+  bool overflow_check = true;
+  uint64_t val;
+  int ret = 0;
+
+  /* addr corresponds to P in the AArch64 ELF document. */
+
+  /* val corresponds to (S + A) in the AArch64 ELF document. */
+
+  val = sym->st_value + rel->r_addend;
+
+  /* Handle the relocation by relocation type */
+
+  switch (ELF64_R_TYPE(rel->r_info))
+    {
+      case R_AARCH64_NONE:
+        {
+          /* No relocation */
+        }
+        break;
+
+      /* Data relocations */
+
+      case R_AARCH64_ABS64:
+        {
+          overflow_check = false;
+          ret = reloc_data(RELOC_OP_ABS, addr, val, 64);
+        }
+        break;
+
+      case R_AARCH64_ABS32:
+        {
+          ret = reloc_data(RELOC_OP_ABS, addr, val, 32);
+        }
+        break;
+
+      case R_AARCH64_ABS16:
+        {
+          ret = reloc_data(RELOC_OP_ABS, addr, val, 16);
+        }
+        break;
+
+      case R_AARCH64_PREL64:
+        {
+          overflow_check = false;
+          ret = reloc_data(RELOC_OP_PREL, addr, val, 64);
+        }
+        break;
+
+      case R_AARCH64_PREL32:
+        {
+          ret = reloc_data(RELOC_OP_PREL, addr, val, 32);
+        }
+        break;
+
+      case R_AARCH64_PREL16:
+        {
+          ret = reloc_data(RELOC_OP_PREL, addr, val, 16);
+        }
+        break;
+
+      case R_AARCH64_MOVW_UABS_G0_NC:
+        {
+          overflow_check = false;
+        }
+        fallthrough;
```

Comment:
I meant something like
```suggestion
        /* fallthrough */
```
but in case if you want to use C23 then most probably should be
```
[[fallthrough]];
```

Reply:
Why can't we just use fallthrougn's implementation directly? need replace it  with a comment?

Reply:
It's enough to add some comment here, other solution(e.g. fallthrough macro or [[fallthrough]]) just make a simple thing complex.

Reply:
Done

Reply:
I think that the comment is enough to understand for reviewer that it is not a missing `break;`, but an intentional decision.